cabal-testsuite/src/Test/Cabal/OutputNormalizer.hs

```diff
@@ -46,6 +46,24 @@ normalizeOutput nenv =
   . resub (posixRegexEscape "tmp/src-" ++ "[0-9]+") "<TMPDIR>"
   . resub (posixRegexEscape (normalizerTmpDir nenv) ++ sameDir) "<ROOT>/"
   . resub (posixRegexEscape (normalizerCanonicalTmpDir nenv) ++ sameDir) "<ROOT>/"
+  . (if buildOS == Windows
+        then
+          -- OK. Here's the deal. In `./Prelude.hs`, `withRepoNoUpdate` sets
+          -- `repoUri` to the tmpdir but with backslashes replaced with
+          -- slashes. This is because Windows treats backslashes and forward
+          -- slashes largely the same in paths, and backslashes aren't allowed
+          -- in a URL like `file+noindex://...`.
+          --
+          -- But that breaks the regexes above, which expect the paths to have
+          -- backslashes.
```

Comment:
The problem mentioned here is in fact true, but that is not the core of the issue. The core of the issue is that `network-uri` cannot parse uris `file+noindex:///C:/...` and this is because file URIs were not defined in the RFC that describes such library ([see](https://www.rfc-editor.org/rfc/rfc3986)).

File uris were described in a later RFC [link](https://www.rfc-editor.org/rfc/rfc8089.html) and in particular Appendix E.2 describes how to parse paths that contain drive letters on Windows.

@hasufell made an excellent job in implementing `file-uri` precisely for this [link](https://hackage.haskell.org/package/file-uri), but we don't use that one. 

We instead use `network-uri` and we use two slashes, so the following:
```
file+noindex://C:/blah
```
results in `C:` being the Auth part of the URI, and `/blah` the actual path. This can be seen in the test pipeline you shared on Matrix, as indeed it mentions paths like `/msys64/tmp/...` which are missing the drive letter.

So in particular we should do the following:
1. When creating a test `local+noindex` repo, use three slashes always. Linux does use three slashes (`"file+noindex://" ++ path` has three slashes), but Windows doesn't.
2. When running `postProcessRepo`, if the scheme is `file+noindex` (a local repo), defer to `file-uri` to resolve it.

There are two other changes needed for the tests to pass:
- an erroneous call to `System.FilePath.Posix.isAbsolute` that should use `System.FilePath.isAbsolute` instead
- When printing the path to the tarFile make sure it uses the right path separator

Then the tests pass without this hacky conversion on the OutputNormalizer.

I am preparing a branch with these.

Reply:
Oh, I remember this issue was litigated to death here: https://github.com/haskell/cabal/pull/9540 URIs is such a cursed topic...